packages/cardano-services/src/Program/programs/providerServer.ts

```diff
@@ -86,6 +86,41 @@ const serviceMapFactory = (options: ServiceMapFactoryOptions) => {
 
   const getEpochMonitor = memoize((dbPool: Pool) => new DbSyncEpochPollService(dbPool, args.epochPollInterval!));
 
+  const getDbSyncStakePoolProvider = withDbSyncProvider((dbPools, cardanoNode) => {
+    if (!genesisData) {
+      throw new MissingProgramOption(ServiceNames.StakePool, ProviderServerOptionDescriptions.CardanoNodeConfigPath);
+    }
+
+    return new DbSyncStakePoolProvider(
+      {
+        paginationPageSizeLimit: args.paginationPageSizeLimit!,
+        responseConfig: { search: { metrics: { apy: !args.disableStakePoolMetricApy } } },
+        useBlockfrost: args.useBlockfrost!
+      },
+      {
+        cache: {
+          db: getDbCache(),
+          healthCheck: healthCheckCache
+        },
+        cardanoNode,
+        dbPools,
+        epochMonitor: getEpochMonitor(dbPools.main),
+        genesisData,
+        logger,
+        metadataService: createHttpStakePoolMetadataService(logger)
+      }
+    );
+  }, ServiceNames.StakePool);
+
+  const getTypeormStakePoolProvider = () => {
+    const entities = [BlockEntity, PoolMetadataEntity, PoolRegistrationEntity, PoolRetirementEntity, StakePoolEntity];
```

Comment:
I wrote a similar line at least twice in some _pg-boss_ related previous PR: I think it's time to refactor this otherwise in the future when we need to change this array we need to apply the same change on some lines spread in the SDK source code.

I suggest to make this `Array` a common value: either hard coded or somehow extracted from [these configuration data](https://github.com/input-output-hk/cardano-js-sdk/blob/ad0a54c09b236034f2824138ffc83a3f6e81cbc8/packages/cardano-services/src/Projection/prepareTypeormProjection.ts#L26-L113).

Reply:
Sounds good to move these "common entities" into a shared package and reuse them whenever it's possible + extend them if needed.

Reply:
They are only common in context of stake pool services. I think the solution here is to resolve using [prepareTypeormProjection](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-services/src/Projection/prepareTypeormProjection.ts) (or add a new util in that file, resolving just the entitites)